src/reader/ft_split_smart.c

```diff
@@ -0,0 +1,120 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ft_split_smart.c                                   :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: al7aro <marvin@42.fr>                      +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2022/09/14 11:32:13 by al7aro            #+#    #+#             */
+/*   Updated: 2022/09/19 12:08:09 by al7aro-g         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdio.h>
+
+#include "libft.h"
+#include "tab.h"
+
+int	is_special(char *str)
+{
+	int	i;
+
+	i = 0;
+	if (*str == '<')
+	{
+		while (*(str + i) == '<')
+			i++;
+		return (i);
+	}
+	if (*str == '>')
+	{
+		while (*(str + i) == '>')
+			i++;
+		return (i);
+	}
+	if (*str == '|')
+		return (1);
+	if (*str == ';')
+		return (1);
+	return (0);
+}
+
+int	is_word(char *str)
+{
+	char	del;
+	int		i;
+
+	i = -1;
+	if (is_special(str))
+		return (is_special(str));
+	del = ' ';
+	while (*(str + ++i) && *(str + i) != del)
```

Comment:
this whole while loop is very hard to read
can you split it to smaller functions like:

`
while (...)
 	{
		if (is_special(str + i, del))
			...
		if (is_excp())
			...
		if (is_del(?))
			...
	}
`